prime-router/src/test/kotlin/fhirengine/translation/hl7/utils/FhirPathUtilsTests.kt

```diff
@@ -88,4 +102,92 @@ class FhirPathUtilsTests {
         assertThat(result[0]).isInstanceOf(ServiceRequest::class.java)
         assertThat((result[0] as ServiceRequest).id).isEqualTo(servRequest.id)
     }
+
+    @Test
+    fun `test evaluateString`() {
+        val bundle = Bundle()
+        bundle.timestamp = Date()
+
+        val observation = Observation()
+        observation.effective = DateTimeType("2015-04-11T12:22:01-04:00")
+        bundle.addEntry().resource = observation
+
+        // Test timestamp of java Date
+        var path = FhirPathUtils.parsePath("Bundle.timestamp")
+        var result = FhirPathUtils.evaluateString(null, bundle, bundle, path!!)
+        assertThat(result).isNotEmpty()
+        assertThat(result).isInstanceOf(String::class.java)
+        assertThat(regex.matches(result)).isTrue()
```

Comment:
I am not sure the regex check is necessary as you are hard coding the pattern in another file.  This may make the maintenance of the pattern a bit more complicated as it now resides in two places.  Perhaps use the terser to make sure it accepts the format?  On top of that, you are only checking the pattern and not the contents (is the time or day in the right place?).  Maybe just test the string returned has the correct value (e.g. 20150411...)